### PR TITLE
fix dosage display and add unit test

### PR DESCRIPTION
**Pull request by jgrant22**
addresses https://github.com/LinuxForHealth/hl7v2-fhir-converter/issues/95

### CODE REVIEW

**Review by cragun47 — APPROVED** [review]
Looks fine.  I would maybe next time tighten up the line spacing in the code.

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java [hunk 1]
```diff
@@ -154,6 +151,40 @@ public void test_ORU_r01_without_status() throws IOException {
     assertThat(status).isEqualTo(DiagnosticReport.DiagnosticReportStatus.UNKNOWN);
   }
 
+  @Test
+  public void test_dosage_output() throws  IOException {
+    String hl7message =
+            "MSH|^~\\&|MyEMR|DE-000001| |CAIRLO|20160701123030-0700||VXU^V04^VXU_V04|CA0001|P|2.6|||ER|AL|||||Z22^CDCPHINVS|DE-000001\r" +
+                    "PID|1||PA123456^^^MYEMR^MR||JONES^GEORGE^M^JR^^^L|MILLER^MARTHA^G^^^^M|20140227|M||2106-3^WHITE^CDCREC|1234 W FIRST ST^^BEVERLY HILLS^CA^90210^^H||^PRN^PH^^^555^5555555||ENG^English^HL70296|||||||2186-5^ not Hispanic or Latino^CDCREC||Y|2\r" +
+                    "PD1|||||||||||02^REMINDER/RECALL - ANY METHOD^HL70215|N|20140730|||A|20140730|\r" +
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/FHIRConverterTest.java]: Suggest just including the segments needed for this function.  Not sure what the minimum is but for sure do not need PD1, NK1, OBX.  Think RXR could also be removed.

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java [hunk 2]
```diff
@@ -154,6 +151,40 @@ public void test_ORU_r01_without_status() throws IOException {
...
+    List<Resource> immunization =
+            e.stream().filter(v -> ResourceType.Immunization == v.getResource().getResourceType())
+                    .map(BundleEntryComponent::getResource).collect(Collectors.toList());
+    assertThat(immunization).hasSize(1);
+
+
+    String s = context.getParser().encodeResourceToString(immunization.get(0));
+    Class<? extends IBaseResource> klass = Immunization.class;
+    Immunization expectDoseQuantity = (Immunization) context.getParser().parseResource(klass, s);
+    assertThat(expectDoseQuantity.hasDoseQuantity()).isTrue();
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:178]: Add tests for the actual value and unit of the doseage

**Review by LisaWellman — CHANGES_REQUESTED** [review]
Please enhance the unit test to check the doseage specifics

**Pull request merged**